Remove backward op for slow dilated 3d convolution

jbschlosser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #70068

Differential Revision: [D33172550](https://our.internmc.facebook.com/intern/diff/D33172550)

jbschlosser committed:
Remove backward op for slow dilated 3d convolution

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33172550).

jbschlosser committed:
Update on "Remove backward op for slow dilated 3d convolution"


Differential Revision: [D33172550](https://our.internmc.facebook.com/intern/diff/D33172550)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33172550).

jbschlosser committed:
Update on "Remove backward op for slow dilated 3d convolution"


Differential Revision: [D33172550](https://our.internmc.facebook.com/intern/diff/D33172550)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33172550).

jbschlosser committed:
Update on "Remove backward op for slow dilated 3d convolution"


Differential Revision: [D33172550](https://our.internmc.facebook.com/intern/diff/D33172550)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33172550).

zou3519 reviewed (approved):
LGTM as long as we change this to use https://github.com/pytorch/pytorch/pull/70332 so we don't end up compiling a bunch of CPU kernels

zou3519 commented on tools/autograd/derivatives.yaml:
```diff
@@ -2028,10 +2028,7 @@
-
-- name: slow_conv_dilated3d_backward(Tensor grad_output, Tensor self, Tensor weight, int[3] kernel_size, int[3] stride, int[3] padding, int[3] dilation, bool[3] output_mask) -> (Tensor grad_input, Tensor grad_weight, Tensor grad_bias)
-  grad_output, self, weight: _convolution_double_backward(grads[0], grads[1], grads[2], grad_output, weight, self, stride, padding, dilation, false, {{0, 0, 0}}, 1, false, false, false, false, grad_input_mask)
+  self, weight, bias: "grad.defined() ? convolution_backward(grad, self, weight, bias->sizes(), stride, padding, dilation, false, std::vector<int64_t>(padding.size(), 0), 1, grad_input_mask) : std::tuple<Tensor, Tensor, Tensor>()"
```
`std::vector<int64_t>(padding.size(), 0)` -> I'm confused, can we pass `padding` directly here?

If not, could we make this a DimVector instead? std::vector is going to use a dynamic allocation

zou3519 commented on tools/autograd/derivatives.yaml:
```diff
@@ -2028,10 +2028,7 @@
-
-- name: slow_conv_dilated3d_backward(Tensor grad_output, Tensor self, Tensor weight, int[3] kernel_size, int[3] stride, int[3] padding, int[3] dilation, bool[3] output_mask) -> (Tensor grad_input, Tensor grad_weight, Tensor grad_bias)
-  grad_output, self, weight: _convolution_double_backward(grads[0], grads[1], grads[2], grad_output, weight, self, stride, padding, dilation, false, {{0, 0, 0}}, 1, false, false, false, false, grad_input_mask)
+  self, weight, bias: "grad.defined() ? convolution_backward(grad, self, weight, bias->sizes(), stride, padding, dilation, false, std::vector<int64_t>(padding.size(), 0), 1, grad_input_mask) : std::tuple<Tensor, Tensor, Tensor>()"
```
nit: /*output_padding*/std::vector<int64_t>(padding.size(), 0) to make it more readable. Also because this is 3d convolution I think you can just pass `{0, 0, 0}`

jbschlosser replied:
Yep sounds good! Will do

jbschlosser committed:
Update on "Remove backward op for slow dilated 3d convolution"


Differential Revision: [D33172550](https://our.internmc.facebook.com/intern/diff/D33172550)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33172550).
